docs/proposals/workflow.md

```diff
@@ -0,0 +1,281 @@
+Workflows (aka [DAG](https://en.wikipedia.org/wiki/Directed_acyclic_graph) workflows
+since _tasks_ are organized in a Directed Acyclic Graph) are ubiquitous
+in modern [job schedulers](https://en.wikipedia.org/wiki/Job_scheduler), see for example:
+
+* [luigi](https://github.com/spotify/luigi)
+* [ozie](http://oozie.apache.org/)
+* [azkaban](https://azkaban.github.io/)
+
+Most of the [job schedulers](https://en.wikipedia.org/wiki/List_of_job_scheduler_software) offer
+workflow functionality to some extent.
+
+
+## Use Cases
+
+* As a user, I want to create a _JobB_ which depends upon _JobA_ running to completion.
+* As a user, I want workflow composability. I want to create a _JobA_ which will be triggered
+as soon as an already running workflow runs to completion.
+* As a user, I want to delete a workflow (eventually cascading to running _tasks_).
+* As a user, I want to debug a workflow (ability to track failure): in case a _task_
+didn't run user should have a way to backtrack the reason of the failure, understanding which
+dependency has not been satisified.
+
+
+## Implementation
+
+This proposal introduces a new REST resource `Workflow`. A `Workflow` is represented as a
+[graph](https://en.wikipedia.org/wiki/Graph_(mathematics)), more specifically as a DAG.
+Vertices of the graph represent steps of the workflow. The workflow steps are represented via a
+`WorkflowStep`<sup>1</sup> resource.
+The edges of the graph represent _dependecies_. To represent edges there is no explicit resource
+- rather they are stored as predecessors in each `WorkflowStep` (i.e. each node).
+The basic idea of this proposal consists in creation of each step postponing execution
+until all predecessors' steps run to completion.
+
+
+### Postponing execution
+
+At the time of writing, to defer execution there are two discussions in the community:
+[#17305](https://github.com/kubernetes/kubernetes/pull/17305): an
+_initializer_ is a dynamically registered object which permits to select a custom controller
+to be applied to a resource. The controller verifies the dependencies.
+The controller checks are applied before the resource is created (even API validated).
+Using a proper controller one may defer creation of the resource until prerequisites
+are satisfied. Even if not completed [#17305](https://github.com/kubernetes/kubernetes/pull/17305)
+already introduces a _dependency_ concept
+([see this comment](https://github.com/kubernetes/kubernetes/pull/17305#discussion_r45007826))
+which could be reused to implement `Workflow`. In
+[#1899](https://github.com/kubernetes/kubernetes/issues/1899):
+some use-cases to wait for specific conditions (`complete`, `ready`) are presented.
+
+
+### Detecting run to completion
+
+To detect run to completion for the resource inside the graph the resource needs to implement
+in `status` the slice of `condition`s. [See](../../docs/devel/api-conventions.md#objects)
+and [#7856](https://github.com/kubernetes/kubernetes/issues/7856).
```

Comment:
Suggestion:
`See [api-convetions](../../docs/devel/api-conventions.md#objects) and [eliminating phases](https://github.com/kubernetes/kubernetes/issues/7856)`
Avoid doing links on see, but rather provide short one or two word description what's under the link. This helps reader to understand what to expect when clicking the link, without it you need to click to know that and we should not force readers to do that.